metacat-common-server/src/main/java/com/netflix/metacat/common/server/util/PoolStatsInterceptor.java

```diff
@@ -58,9 +61,9 @@ public void disconnected(final ConnectionPool parent, final PooledConnection con
 
     private void publishMetric(final ConnectionPool parent) {
         if (parent != null && metricNameTotal != null) {
-            DynamicGauge.set(metricNameTotal, parent.getSize());
-            DynamicGauge.set(metricNameActive, parent.getActive());
-            DynamicGauge.set(metricNameIdle, parent.getIdle());
+            registry.gauge(registry.createId(metricNameTotal), parent.getSize());
```

Comment:
Spectator supports an active and passive gauge. Active gauges are similar to counters and other types where the user calls set when something happens (`publishMetric` in this case). The passive case you provide the number reference or a function to compute the number and spectator will poll it in the background until that reference is garbage collected. Based on the usage pattern of the previous you would want the activity based notion here. So it would look like:

```java
registry.gauge(registry.createId(metricNameTotal)).set(parent.getSize());
```

It is unfortunate they appear so similar in the api, but it has to do with how spectator evolved and there are some backwards compatibility constraints at this point.

You can cache the id, but in this case since the id doesn't need to be updated it would be better to just cache the gauge reference. Something like:

```java
private Gauge totalGauge;
...

private void publishMetric(final ConnectionPool parent) {
  ...
  totalGauge.set(parent.getSize());
  ...
}

public void setMetricName(final String metricName) {
  this.totalGauge = registry.gauge(registry.createId(LogConstants.GaugeConnectionsTotal + "." + metricName));
  ...
}
```